compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -36,14 +40,32 @@ namespace iree_compiler {
 
 namespace {
 
+// Transform gpu.barrier -> amdgpu.lds_barrier
+struct RewriteBarriers  : public OpRewritePattern<gpu::BarrierOp> {
+  using OpRewritePattern<gpu::BarrierOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(gpu::BarrierOp op,
+                                PatternRewriter &rewriter) const override {
+    OpBuilder::InsertionGuard guard(rewriter);
+    rewriter.setInsertionPoint(op);
```

Comment:
I wouldn't expect you to need to set the insertion point like this; isn't it already set to immediately after the barrier op?